ring/basic_lifecycler.go

```diff
@@ -514,6 +534,31 @@ func (l *BasicLifecycler) changeState(ctx context.Context, state InstanceState)
 	return err
 }
 
+func (l *BasicLifecycler) changeReadOnlyState(ctx context.Context, readOnly bool) error {
+	err := l.updateInstance(ctx, func(_ *Desc, i *InstanceDesc) bool {
+		// No-op if the state hasn't changed.
+		if i.ReadOnly == readOnly {
+			return false
+		}
+
+		if readOnly {
+			i.ReadOnly = true
+			i.ReadOnlyUpdatedTimestamp = time.Now().Unix()
+		} else {
+			i.ReadOnly = false
+			i.ReadOnlyUpdatedTimestamp = 0
```

Comment:
In the `Lifecycler` implementation, this is also set to the `time.Now()` when we set `ReadOnly = false`. Reason is that its meaning is the timestamp of when the flag change the last time (if it ever changed). I think we should keep it consistent here.

Reply:
```suggestion
			i.ReadOnlyUpdatedTimestamp = time.Now().Unix()
```